Move theming support from Beta to Main

jordanjones243 opened the pull request:
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Resolves:** #57 

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is and anything that describes the before/after state of the project._

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)


## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

<!-- Generated by sourcery-ai[bot]: start summary -->

## Summary by Sourcery

Move theming support for the auro-banner component from beta to main, enabling restyling through design tokens. Refactor color tokens to align with tier 3 standards and update documentation to reflect these changes.

New Features:
- Introduce theme support for the auro-banner component, allowing it to be restyled using new design tokens.

Enhancements:
- Refactor color tokens to use tier 3 tokens for improved consistency and maintainability.

Documentation:
- Update documentation to include information on the new theme support and how to use design tokens for restyling the auro-banner component.

<!-- Generated by sourcery-ai[bot]: end summary -->

jordanjones243 committed:
build: update design tokens and all other dependencies

jordanjones243 committed:
feat: refactor color tokens with tier 3 tokens #57

jason-capsule42 committed:
perf: update dependencies

semantic-release-bot committed:
chore(release): 2.1.0-beta.1 [skip ci]

# [2.1.0-beta.1](https://github.com/AlaskaAirlines/auro-banner/compare/v2.0.7...v2.1.0-beta.1) (2024-07-09)

### Features

* refactor color tokens with tier 3 tokens [#57](https://github.com/AlaskaAirlines/auro-banner/issues/57) ([1b104b7](https://github.com/AlaskaAirlines/auro-banner/commit/1b104b7be217d4dafca07c8ac42b04e40c561c3e))

### Performance Improvements

* update dependencies ([8f451b7](https://github.com/AlaskaAirlines/auro-banner/commit/8f451b7fbf846027cebe4552fd062583afa039cb))

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/AlaskaAirlines/auro-banner?pullRequest=63) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/AlaskaAirlines/auro-banner?pullRequest=63) before we can accept your contribution.<br/>**2** out of **3** committers have signed the CLA.<br/><br/>:white_check_mark: jordanjones243<br/>:white_check_mark: jason-capsule42<br/>:x: semantic-release-bot<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/AlaskaAirlines/auro-banner?pullRequest=63) it.</sub>

jordanjones243 committed:
perf: update dependencies and lint configuration

jordanjones243 committed:
fix: resolve minor color theming issues #57

jordanjones243 committed:
feat: use custom registered auro-header

jordanjones243 committed:
feat: add attribute as tag name when custom registered

jordanjones243 committed:
perf: refactor custom component registration config

jordanjones243 committed:
docs: remove beta releases from CHANGELOG

jordanjones243 commented on src/tokens.scss:
```diff
@@ -0,0 +1,10 @@
+@import "./../node_modules/@aurodesignsystem/design-tokens/dist/tokens/SCSSVariables";
+
+:host {
+  --ds-auro-banner-text-color: var(--ds-color-text-primary-default, #{$ds-color-text-primary-default});
```
Not adding a comment at this time.

jordanjones243 commented on docs/partials/api.md:
```diff
@@ -158,3 +158,10 @@ The following example illustrates a `<auro-banner>` custom element with the `rou
 <!-- AURO-GENERATED-CONTENT:END -->
```
It is indeed rendered on the API page.

semantic-release-bot committed:
chore(release): 2.1.0-beta.2 [skip ci]

# [2.1.0-beta.2](https://github.com/AlaskaAirlines/auro-banner/compare/v2.1.0-beta.1...v2.1.0-beta.2) (2024-09-17)

### Bug Fixes

* resolve minor color theming issues [#57](https://github.com/AlaskaAirlines/auro-banner/issues/57) ([fe2533d](https://github.com/AlaskaAirlines/auro-banner/commit/fe2533dc6662ffa996b5fed0a1767c446e1324c7))

### Features

* add attribute as tag name when custom registered ([31a0765](https://github.com/AlaskaAirlines/auro-banner/commit/31a07652b861f5d437b04b6ed5c1af3909a5bf76))
* use custom registered auro-header ([0f61d33](https://github.com/AlaskaAirlines/auro-banner/commit/0f61d334412f319424aa6998845c194ada31c26b))

### Performance Improvements

* refactor custom component registration config ([f9907fa](https://github.com/AlaskaAirlines/auro-banner/commit/f9907faaf66bd7d918f0d6599ce51c363fa79d52))
* update dependencies and lint configuration ([2463c42](https://github.com/AlaskaAirlines/auro-banner/commit/2463c422145950cedb409adef15afe8bf4af5dfd))

jason-capsule42 commented on CHANGELOG.md:
Remove the beta entries from the changelog

jordanjones243 committed:
docs: remove beta release from changelog

jordanjones243 committed:
perf: update dependencies

BREAKING CHANGE: trigger major release for color theme support #57

semantic-release-bot committed:
chore(release): 3.0.0-beta.1 [skip ci]

# [3.0.0-beta.1](https://github.com/AlaskaAirlines/auro-banner/compare/v2.1.0-beta.2...v3.0.0-beta.1) (2024-09-24)

### Performance Improvements

* update dependencies ([757c078](https://github.com/AlaskaAirlines/auro-banner/commit/757c078dcea8e6cd88b4f61c93f7fa7ddaaed23f))

### BREAKING CHANGES

* trigger major release for color theme support #57

jordanjones243 committed:
docs: remove beta release from changelog

blackfalcon commented:
:tada: This PR is included in version 3.0.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@aurodesignsystem/auro-banner/v/3.0.0)
- [GitHub release](https://github.com/AlaskaAirlines/auro-banner/releases/tag/v3.0.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
